subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -539,4 +541,48 @@ public String determineName(Configuration c) {
      */
     boolean isCanBeResolved();
 
+    /**
+     * This method can be used to determine if a configuration of a Gradle core plugin is deprecated
+     * for declaring dependencies. If it is the case, the method returns a list, if not it returns 'null'.
```

Comment:
Not too fond of this contract.
What about:
> If it is the case, the method returns a non-empty list. If not it returns an empty list.

I really prefer having not nullable APIs when we can.
I also thought about throwing, for `getConsumptionAlternatives` and `getResolutionAlternatives` when the matching flag is `false` but I believe this would only complicate usage while not providing much if we never return `null` anyway.

Reply:
Lets discuss this offline.

Reply:
We decided to put the public query methods into a separate interface

`Configuration.getDeprecationInformation()`